### ISSUE TITLE
Flickering Screen / Crashing / Unresponsive To Taps / Keyboard Fllickering

### ISSUE THREAD

**Issue by FroggMaster** [MuntashirAkon/AppManager#1317]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

- Upon launching App Manager the screen is flickering
- The application takes several taps before it will respond to input
- The keyboard will appear then disappear over and over again making it impossible to type
- The app will frequently suspend itself to the background or close but you can simply re-open it on the screen it closed on. 

### To Reproduce

1. Install App Manager v3.1.5 on Google Pixel 6 Pro running Android 14: Firmware build: AP1A.240305.019.A1, Mar 2024

### Expected behavior

For none of the above issues to be presented; I would expect the app to be responsive, not close, the screen not to flicker and the keyboard to be functional.

### Device info

- Device:  Google Pixel 6 Pro (raven)
- OS Version: Android 14 / AP1A.240305.019.A1, Mar 2024
- App Manager Version: 3.1.5
- Mode: Root / NonRoot (issue is displayed before root permissions are granted, but persists after root is provided.)

**Comment by MuntashirAkon** [comment]
This is not an Android 14 issue since I run Android 14 myself. However, this could be a display-related issue which I cannot fix without getting hands on the device itself.

**Comment by FroggMaster** [comment]
It's unfortunate because the bug makes the app essentially unable to be used on my device. If you need additional information or someone to test any updates that might resolve this issue on the device, I'd be happy to help. 

Unfortunately, I know how difficult it is to attempt to troubleshoot and resolve something when you're not able to replicate the problem,.

**Comment by MuntashirAkon** [comment]
You can share a video of the issue, recorder from another device. Make sure to be quite thorough so that I may be able to guess why it may be happening. Note that, no version or breaking changes have been introduced in v3.1.5. So, it should be more or less identical to v3.1.4 except a few internal changes. So, if v3.1.4 works but v3.1.5 does not, you have to report this as well. Also, make sure you use the release builds (not debug builds) for this test. Debug builds have extensive logging enabled which may result in more work in the UI/Main thread and cause similar flickering (assuming you've got an amoled or similar display).

**Comment by MuntashirAkon** [comment]
Also, make sure there's no apps interfering with App Manager via accessibility. Revoke the accessibility permission from all apps before doing the above.

**Comment by FroggMaster** [comment]
This issue was occurring for both versions for me both 3.1.4 and 3.1.5. However; I think this may have been a weird glitch with my device. I haven't changed anything but I'm not able to re-create the issue anymore. 

I had reinstalled the application and rebooted multiple times previously and the problem continued. I reinstalled again to attempt to re-create the problem but the app is now working, no flickering, no randomly sending to background and the keyboard is responsive.

**Comment by MuntashirAkon** [comment]
This could be due to dex optimisation. If this happens again, please perform a dexopt operation on App Manager via terminal and see if it stops after that.

**Comment by Terrance** [comment]
I've just installed the 4.0 beta release (after uninstalling 3.x, as I previously installed via F-Droid with a different signature) and I'm also seeing the behaviour described here.

Logs: 📎 [29_04-18-53-34_533.log](https://github.com/MuntashirAkon/AppManager/files/15153477/29_04-18-53-34_533.log)

- there's a lot of `onFocusEvent` spam

https://github.com/MuntashirAkon/AppManager/assets/4025899/2caddc07-cfff-4f6d-9b6b-de5cf65551b5

- Android top bar is flickering, I'm guessing due to switching between app and system theme colours
- some of the back swipes aren't even visible in the video, not sure if that's down to the frame rate going choppy from the lag or because the inputs are actually cancelled before displaying the on-screen hint
- most taps of menu options or the top-left back arrow button are similarly ignored

**Comment by Terrance** [comment]
> If this happens again, please perform a dexopt operation on App Manager via terminal and see if it stops after that.

What exactly do I need to run in order to do this?  Internet searches are leading me to `pm`, but...

```
OnePlusNordCE:/ $ pm compile io.github.muntashirakon.AppManager
Error: Failed to cpmpile !
OnePlusNordCE:/ $ pm force-dex-opt io.github.muntashirakon.AppManager
                                                                                                                                                                                                                                                                          Exception occurred while executing 'force-dex-opt':
java.lang.SecurityException: forceDexOpt
        at com.android.server.pm.PackageManagerServiceUtils.enforceSystemOrRoot(PackageManagerServiceUtils.java:1313)
        at com.android.server.pm.DexOptHelper.forceDexOpt(DexOptHelper.java:536)
        at com.android.server.pm.IPackageManagerBase.forceDexOpt(IPackageManagerBase.java:300)
        at com.android.server.pm.PackageManagerShellCommand.runForceDexOpt(PackageManagerShellCommand.java:1995)
        at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:264)
        at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
        at android.os.ShellCommand.exec(ShellCommand.java:38)
        at com.android.server.pm.PackageManagerService$IPackageManagerImpl.onShellCommand(PackageManagerService.java:6362)
        at android.os.Binder.shellCommand(Binder.java:1073)
        at android.os.Binder.onTransact(Binder.java:901)
        at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:4313)
        at com.android.server.pm.PackageManagerService$IPackageManagerImpl.onTransact(PackageManagerService.java:6346)
        at android.os.Binder.execTransactInternal(Binder.java:1331)
        at android.os.Binder.execTransact(Binder.java:1268)
```

**Comment by wojnilowicz** [comment]
I've also seen this issue on LineageOS 21, but reinstalling the AppManager 3.1.6 from F-Droid helped.

**Comment by Terrance** [comment]
For me, this appears to be down to denying notification permissions -- granting that permission and restarting the app works; denying and restarting again causes the flickering and lag to return.

On a fresh install, both the first and second launches show a request for the permission.  At a guess, the app is getting stuck in a loop trying to show the permission request dialog on subsequent launches, after Android starts implicitly denying the requests?

**Comment by MuntashirAkon** [comment]
> On a fresh install, both the first and second launches show a request for the permission. At a guess, the app is getting stuck in a loop trying to show the permission request dialog on subsequent launches, after Android starts implicitly denying the requests?

Your issue is identical to #1173 and should now be fixed (via 93ce56cc89f9fc4fd109098370b933ff1f1bb430).
